bundles/edu.kit.kastel.sdq.eclipse.grading.view/src/edu/kit/kastel/eclipse/grading/view/commands/AddAnnotationDialog.java

```diff
@@ -0,0 +1,158 @@
+public class AddAnnotationDialog extends Dialog {
+	private final AssessmentViewController controller;
+	private TableViewer displayList;
+	private AnnotationFilter filter;
+	
+	private IMistakeType selectedMistake;
+	private boolean customMessageWanted;
+
+	public AddAnnotationDialog(Shell parentShell, AssessmentViewController controller) {
+		super(parentShell);
+		this.controller = controller;
+	}
+	
+	public Optional<IMistakeType> getSelectedMistake() {
+		return Optional.ofNullable(this.selectedMistake);
+	}
+	
+	public boolean isCustomMessageWanted() {
+		return this.customMessageWanted;
+	}
+	
+	@Override
+	protected Control createDialogArea(Composite parent) {
+		Composite container = (Composite) super.createDialogArea(parent);
+		container.setSize(SWT.DEFAULT, 200);
+		
+		GridLayout layout = new GridLayout();
+		container.setLayout(layout);
+		
+		this.createSearchField(container);
+		this.createAnnotationList(container);
+		
+		return container;
+	}
+	
+	@Override
+	protected void configureShell(Shell shell) {
+		super.configureShell(shell);
+		shell.setText("Add Annotation");
+	}
+	
+	@Override
+	protected void createButtonsForButtonBar(Composite parent) {
+		// Removes the ok and close buttons
+	}
+	
+	private void updateDisplayList(String filter) {
+		this.filter.setFilterString(filter);
+		this.displayList.refresh();
+		this.displayList.getTable().setSelection(0);
+	}
+	
+	private void createSearchField(Composite container) {
+		Text inputField = new Text(container, SWT.SINGLE | SWT.BORDER);
+		inputField.addTraverseListener(e -> {
+			if (e.detail == SWT.TRAVERSE_RETURN) {
+				processMistakeSelection(isShiftPressed(e.stateMask));
+				this.okPressed();
+			} else if (e.detail == SWT.TRAVERSE_ESCAPE) {
+				this.cancelPressed();
+			}
+		});
+		inputField.addKeyListener(new KeyAdapter() {
+			@Override
+			public void keyPressed(KeyEvent e) {
+				int index = displayList.getTable().getSelectionIndex();
+				if (e.keyCode == SWT.ARROW_DOWN) {
+					displayList.getTable().select(index + 1);
+				} else if (e.keyCode == SWT.ARROW_UP) {
+					displayList.getTable().select(index - 1);
+				}
+			}
+		});
+		inputField.addModifyListener(e -> updateDisplayList(inputField.getText()));
+		inputField.setFocus();
+		
+		GridData gridData = new GridData();
+		gridData.verticalAlignment = GridData.FILL;
+		gridData.grabExcessHorizontalSpace = true;
+		gridData.horizontalAlignment = GridData.FILL;
+		inputField.setLayoutData(gridData);
+	}
+	
+	private void createAnnotationList(Composite container) {
+		// List (-> table) of mistake types
+		this.displayList = new TableViewer(container, SWT.SINGLE | SWT.H_SCROLL | SWT.V_SCROLL);
+		this.displayList.getTable().setHeaderVisible(false);
+		this.displayList.getTable().setLinesVisible(false);
+		this.displayList.setLabelProvider(new LabelProvider() {
+			@Override
+			public String getText(Object item) {
+				return ((IMistakeType) item).getButtonText();
```

Comment:
Maybe the label could contain (parts) of the description of the button? What do you think?

Reply:
Changed in the latest commit. The dialog size is subject to further tweaking :D
![grafik](https://user-images.githubusercontent.com/22750108/195727224-2a627b6d-d3d4-4070-be2c-533a519e5f7b.png)